olive/model/__init__.py

```diff
@@ -513,6 +588,12 @@ def load_model(self, rank: int = None) -> torch.nn.Module:
             else:
                 raise ValueError(f"Unsupported model file format: {self.model_file_format}")
 
+        # we only have peft adapters for now
+        if self.adapter_path:
+            from peft import PeftModel
+
+            model = PeftModel.from_pretrained(model, self.adapter_path)
```

Comment:
are there any common way to handle this? This seems a little bit hacky. 

For these model loading need special handling, could we leverage the customer model loader by preparing couple of built-in loader functions?

Reply:
Adapters are gaining wider adaption for LLM use cases.

We need to be prepared to support following scenarios in Olive:
- [PT Model, Adapter Weights] -> [PT Pass] -> [Merge Adapter weights] -> [ONNX Conversion] -> [ONNX Model] -> ...
- [PT Model, Adapter Weights] -> [PT Pass] -> [ONNX Conversion] -> [ONNX Model, ONNX Adapter Weights] -> ...
- [PT Model] -> [PT Pass] -> [PT Model, Adapter Weights] -> ....
- [ONNX Model] -> [ONNX Pass] -> [ONNX Model, Adapter Weights] -> ...